test/io/test_octree_compression.cpp

```diff
@@ -48,39 +48,34 @@ int total_runs = 0;
 char* pcd_file;
 
 #define MAX_POINTS 10000.0
-#define MAX_XYZ 1024.0
 #define MAX_COLOR 255
-#define NUMBER_OF_TEST_RUNS 2
+#define NUMBER_OF_TEST_RUNS 3
 
 TEST (PCL, OctreeDeCompressionRandomPointXYZRGBA)
 {
   srand(static_cast<unsigned int> (time(NULL)));
-
+  for (const double MAX_XYZ : {1.0, 1024.0}) { // Small clouds, large clouds
     // iterate over all pre-defined compression profiles
-  for (int compression_profile = pcl::io::LOW_RES_ONLINE_COMPRESSION_WITHOUT_COLOR;
-    compression_profile != pcl::io::COMPRESSION_PROFILE_COUNT; ++compression_profile) {
-    // instantiate point cloud compression encoder/decoder
-    pcl::io::OctreePointCloudCompression<pcl::PointXYZRGBA> pointcloud_encoder((pcl::io::compression_Profiles_e) compression_profile, false);
-    pcl::io::OctreePointCloudCompression<pcl::PointXYZRGBA> pointcloud_decoder;
-    pcl::PointCloud<pcl::PointXYZRGBA>::Ptr cloud_out(new pcl::PointCloud<pcl::PointXYZRGBA>());
-    // iterate over runs
-    for (int test_idx = 0; test_idx < NUMBER_OF_TEST_RUNS; test_idx++, total_runs++)
-    {
-      try
+    for (int compression_profile = pcl::io::LOW_RES_ONLINE_COMPRESSION_WITHOUT_COLOR;
+      compression_profile != pcl::io::COMPRESSION_PROFILE_COUNT; ++compression_profile) {
+      // instantiate point cloud compression encoder/decoder
+      pcl::io::OctreePointCloudCompression<pcl::PointXYZRGBA> pointcloud_encoder((pcl::io::compression_Profiles_e) compression_profile, false);
+      pcl::io::OctreePointCloudCompression<pcl::PointXYZRGBA> pointcloud_decoder;
+      pcl::PointCloud<pcl::PointXYZRGBA>::Ptr cloud_out(new pcl::PointCloud<pcl::PointXYZRGBA>());
+      // iterate over runs
+      for (int test_idx = 0; test_idx < NUMBER_OF_TEST_RUNS; test_idx++, total_runs++)
       {
-        int point_count = MAX_POINTS * rand() / RAND_MAX;
-        if (point_count < 1)
-        { // empty point cloud hangs decoder
-          total_runs--;
-          continue;
-        }
+        int point_count;
+        do { // empty point cloud hangs decoder
+          point_count = MAX_POINTS * rand() / RAND_MAX;
+        } while (point_count < 1);
```

Comment:
```suggestion
        // empty point cloud hangs decoder
        const int point_count = 1 + MAX_POINTS * rand() / RAND_MAX;
```

Reply:
Can be changed 3 places.

Reply:
Thanks, that is much nicer. I used `MAX_POINTS - 1` though, to keep the correct upper limit